plugins/ecosystem/src/components/DomainExplorerPage/DomainExplorerPage.tsx

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { DomainEntity } from '@backstage/catalog-model';
+import {
+  configApiRef,
+  Content,
+  ContentHeader,
+  EmptyState,
+  Header,
+  Page,
+  Progress,
+  SupportButton,
+  useApi,
+} from '@backstage/core';
+import { catalogApiRef } from '@backstage/plugin-catalog';
+import { Button } from '@material-ui/core';
+import React from 'react';
+import { useAsync } from 'react-use';
+import { DomainExplorer } from '../DomainExplorer/DomainExplorer';
+
+export const DomainExplorerPage = () => {
+  const configApi = useApi(configApiRef);
+  const organizationName =
+    configApi.getOptionalString('organization.name') ?? 'Backstage';
+  const catalogApi = useApi(catalogApiRef);
+  const { value: entities, loading } = useAsync(async () => {
+    const response = await catalogApi.getEntities({
+      filter: { kind: 'domain' },
+    });
+
+    return response.items as DomainEntity[];
+  }, [catalogApi]);
+
+  return (
+    <Page themeId="home">
+      <Header
+        title={`Explore the ${organizationName} ecosystem`}
+        subtitle="Discover the domains in your ecosystem"
+      />
+      <Content>
+        <ContentHeader title="Domains">
+          <SupportButton>Discover the domains in your ecosystem.</SupportButton>
+        </ContentHeader>
+        {loading && <Progress />}
+        {!loading && (!entities || entities.length === 0) && (
+          <EmptyState
+            missing="info"
+            title="No domains to display"
+            description={`You haven't added any domains yet.`}
+            action={
+              <Button
+                variant="contained"
+                color="primary"
+                href="https://backstage.io/docs/features/software-catalog/descriptor-format#kind-domain"
```

Comment:
For an entire Backstage instance to lack domains, I wonder if the lack of domains is less about the mechanics of how to define the Kind, but more a lack of understanding how the model helps map to business benefits?  I might wonder if this link could be more useful being to the model info:
https://backstage.io/docs/features/software-catalog/system-model#ecosystem-modeling

And then maybe what's missing is links from the model page TO the next level of detail, e.g. to the `Kind:domain` and `Kind:system` descriptors?

Reply:
Yeah we could switch that link out

> And then maybe what's missing is links from the model page TO the next level of detail, e.g. to the Kind:domain and Kind:system descriptors?

Yeah, but I don't think that it should be part of this PR. That can be a separate contribution.

Reply:
100% agree, didn't mean to infer to block this PR. :)